src/mecca/platform/os/darwin/package.d

```diff
@@ -0,0 +1,141 @@
+    SIGURG = 16, /// urgent condition on IO channel
+    SIGSTOP = 17, /// sendable stop signal not from tty
+    SIGTSTP = 18, /// stop signal from tty
+    SIGCONT = 19, /// continue a stopped process
+    SIGCHLD = 20, /// to parent on child stop or exit
+    SIGTTIN = 21, /// to readers pgrp upon background tty read
+    SIGTTOU = 22, /// like TTIN for output if (tp->t_local&LTOSTOP)
+    SIGIO = 23, /// input/output possible signal
+    SIGXCPU = 24, /// exceeded CPU time limit
+    SIGXFSZ = 25, /// exceeded file size limit
+    SIGVTALRM = 26, /// virtual time alarm
+    SIGPROF = 27, /// profiling time alarm
+    SIGWINCH = 28, /// window size changes
+    SIGINFO = 29, /// information request
+    SIGUSR1 = 30, /// user defined signal 1
+    SIGUSR2 = 31 /// user defined signal 2
+}
+
+/**
+ * Represents the ID of a thread.
+ *
+ * This type is platform dependent.
+ */
+alias ThreadId = ulong;
+
+__gshared static immutable BLOCKED_SIGNALS = [
+    OSSignal.SIGHUP, OSSignal.SIGINT, OSSignal.SIGQUIT,
+    //OSSignal.SIGILL, OSSignal.SIGTRAP, OSSignal.SIGABRT,
+    //OSSignal.SIGBUS, OSSignal.SIGFPE, OSSignal.SIGKILL,
+    //OSSignal.SIGUSR1, OSSignal.SIGSEGV, OSSignal.SIGUSR2,
+    OSSignal.SIGPIPE, OSSignal.SIGALRM, OSSignal.SIGTERM,
+    //OSSignal.SIGSTKFLT, OSSignal.SIGCONT, OSSignal.SIGSTOP,
+    OSSignal.SIGCHLD, OSSignal.SIGTSTP, OSSignal.SIGTTIN,
+    OSSignal.SIGTTOU, OSSignal.SIGURG, OSSignal.SIGXCPU,
+    OSSignal.SIGXFSZ, OSSignal.SIGVTALRM, OSSignal.SIGPROF,
+    OSSignal.SIGWINCH, OSSignal.SIGIO,
+    //OSSignal.SIGSYS,
+];
+
+extern (C) private int pthread_threadid_np(pthread_t, ulong*) nothrow;
+
+/// Returns: the current thread ID
+ThreadId currentThreadId() @system nothrow
+{
+    import mecca.lib.exception : ASSERT;
+
+    enum assertMessage = "pthread_threadid_np failed, should not happen";
+
+    ulong threadId;
+    ASSERT!"assertMessage"(pthread_threadid_np(null, &threadId) == 0);
+
+    return threadId;
+}
+
+enum O_CLOEXEC = 0x1000000;
+enum F_DUPFD_CLOEXEC = 67;
+
+// this does not exist on Darwin
+enum EREMOTEIO = -1;
+
+// pipe2 does not exist on Darwin so we're ignoring the flag and fallback to the
+// regular `pipe` function
+extern(C) private int pipe2(ref int[2] pipefd, int) nothrow @trusted @nogc
+{
+    import core.sys.posix.unistd : pipe;
+    return pipe(pipefd);
+}
+
+enum ITIMER_REAL = 0;
+
+private extern(C) nothrow @system @nogc
+{
+    import core.sys.posix.pthread: pthread_mutex_t, pthread_mutexattr_t;
+
+    // these are not marked as @nogc in some versions of druntime
+    int pthread_mutex_init(pthread_mutex_t*, pthread_mutexattr_t*);
+    int pthread_mutex_destroy(pthread_mutex_t*);
+    int pthread_mutex_lock(pthread_mutex_t*);
+    int pthread_mutex_unlock(pthread_mutex_t*);
+}
+
+struct PthreadMutex
+{
+    import mecca.lib.exception : ASSERT;
+
+    pthread_mutex_t mtx;
+
+    void open()
+    {
+        ASSERT!"pthread_mutex_init"(pthread_mutex_init(&mtx, null) == 0);
```

Comment:
If you keep the `init`/`destroy` calls - they're Linux-compatible too, so no need to split the code.

Reply:
Not sure I follow. I thought I was supposed to remove these and the whole `PthreadMutex` wrapper and use `PTHREAD_MUTEX_INITIALIZER` instead?

Reply:
Sure. If it isn't possible for some reason, then this is a simple alternative

Reply:
Sure. If it isn't possible for some reason, then this is a simple alternative